crates/common_utils/src/types.rs

```diff
@@ -483,6 +484,20 @@ impl Sub for MinorUnit {
     }
 }
 
+impl Mul for MinorUnit {
+    type Output = Self;
+
+    fn mul(self, a2: Self) -> Self::Output {
+        Self(self.0 * a2.0)
+    }
+}
+
+impl Sum for MinorUnit {
```

Comment:
add is already there why is Sum needed?

Reply:
.sum() function is required to accumulate the elements of an iterator for OrderDetails.